database/cassandra/cassandra.go

```diff
@@ -226,16 +230,24 @@ func (c *Cassandra) Run(migration io.Reader) error {
 }
 
 func (c *Cassandra) SetVersion(version int, dirty bool) error {
-	query := `TRUNCATE "` + c.config.MigrationsTable + `"`
-	if err := c.session.Query(query).Exec(); err != nil {
-		return &database.Error{OrigErr: err, Query: []byte(query)}
+	if previous, _, err := c.Version(); err == nil {
+		if previous != database.NilVersion {
+			// DELETE instead of TRUNCATE because AWS Keyspaces does not support it
+			// see: https://docs.aws.amazon.com/keyspaces/latest/devguide/cassandra-apis.html
+			query := `DELETE FROM "` + c.config.MigrationsTable + `" WHERE version = ?`
```

Comment:
This only removes the previous version and not all versions. Having multiple versions will cause issues since selecting the current versions only selects one version without any ordering.

We can either:
1. Delete all rows and keep the current behavior
2. Update `Version()` to select the latest version
3. ??? - Other backwards compatible suggestions?

Reply:
I believe this causing the tests to fail. You can test locally by running `go test -v ./database/cassandra`. You'll need docker installed.